src/main/java/uk/gov/ch/oauth/Oauth2.java

```diff
@@ -118,10 +129,10 @@ public boolean oauth2VerifyNonce(final String nonce, final String sessionNonce)
      *
      * @return The Nonce String from within the session or null if not found.
      */
-    public String getSessionNonce(final Session chSession) {
+    private String getSessionNonce() {
         String oauth2Nonce = null;
         try {
-            final Map<String, Object> data = chSession.getData();
+            final Map<String, Object> data = sessionUtils.getSessionDataFromContext();
```

Comment:
This was changed as the session fails to update when getting the session this way. Have we tested to see whether the nonce is removed from the session after signing in through to signing out.

Reply:
As we showed via inspection. This does infact work.